reconcile/status_board.py

```diff
@@ -305,17 +304,21 @@ def desired_abstract_status_board_map(
         """
         desired_abstract_status_board_map: dict[str, AbstractStatusBoard] = {}
         for product, apps in desired_product_apps.items():
+            current_product = current_status_board_map.get(product)
+            product_id = current_product.id if current_product else None
             desired_abstract_status_board_map[product] = Product(
-                name=product, fullname=product, applications=[], metadata={}
+                id=product_id, name=product, fullname=product, applications=[]
             )
             for a in apps:
                 key = f"{product}/{a}"
+                current_app = current_status_board_map.get(key)
+                app_id = current_app.id if current_app else None
                 desired_abstract_status_board_map[key] = Application(
+                    id=app_id,
                     name=a,
                     fullname=key,
                     services=[],
-                    product=desired_abstract_status_board_map[product],
-                    metadata={},
+                    product=desired_abstract_status_board_map.get(product),
```

Comment:
please rename current local var `product` to `product_name`, and extract a var `product` to hold instance of `Product`, use it to populate map and application, to avoid warning `Expected type 'Product', got 'AbstractStatusBoard | None' instead`:

```python
for product_name, apps in desired_product_apps.items():
    product = Product(...)
    desired_abstract_status_board_map[product_name] = product
    for a in apps:
        ...
        = Application(..., product=product)
```

Reply:
Done.